src/terminal_box.rs

```diff
@@ -410,7 +410,7 @@ where
                                 let bottom_offset = style_line_height * 2.0;
 
                                 let full_width = self.end_x - self.start_x;
-                                let mut accu_width = 0.0;
+                                let mut accu_width = -(self.start_x % 4.0);
                                 let mut dot_width = 2.0f32.min(full_width - accu_width);
```

Comment:
With negative `accu_width`, the `.min()` call would not limit the width of the underline to the span.

Same with the other styles below.

Reply:
I reworked them and unified into one function for drawing repeating pattern underlines. I also make sure thatit starts and stops within the span :+1: